src/merkle_proof/proof.rs

```diff
@@ -26,10 +26,23 @@ pub fn merkle_opening_gadget(
     proven_leaf: Variable,
     proven_root: BlsScalar,
 ) -> Variable {
+    // Loading the level leaves into the composer and return references to the allocated leaves
+    // (Variables) inside of an array of length = WIDTH so that it can be hashed with the
+    // Hades252::GadgetStrategy::perm().
+    let allocate_poseidon_level =
+        |composer: &mut StandardComposer,
+         level: &PoseidonLevel,
+         lvl_vars: &mut [Variable; WIDTH]| {
+            level
+                .leaves
+                .iter()
+                .zip(lvl_vars.iter_mut())
+                .for_each(|(leaf, var)| *var = composer.add_input(*leaf));
+        };
     // Generate and constraint zero.
     let zero = composer.add_witness_to_circuit_description(BlsScalar::zero());
     // Allocate space for each level Variables that will be generated.
-    let mut lvl_vars = [zero; WIDTH];
+    let mut lvl_vars = unsafe { [std::mem::zeroed(); WIDTH] };
```

Comment:
I assume there is a reason for this change? Could you elaborate (and maybe add it to the comment)?

Reply:
Reverted.